web/hooks/use-group-supabase.ts

```diff
@@ -65,15 +79,17 @@ export function useRealtimeGroupContractIds(groupId: string) {
   return rows?.map((r) => r.contract_id)
 }
 
-export const useGroupsWithContract = (contract: Contract) => {
+export const useGroupsWithContract = (
+  contract: Contract | undefined | null
+) => {
   const [groups, setGroups] = useState<Group[]>()
 
-  useEffect(() => {
-    if (contract.groupSlugs)
+  useEffectCheckEquality(() => {
+    if (contract && contract.groupSlugs)
       listGroupsBySlug(uniq(contract.groupSlugs)).then((groups) =>
         setGroups(filterDefined(groups))
       )
-  }, [contract.groupSlugs])
+  }, [contract])
```

Comment:
You want this to update whenever any property on contract changes? Seems unlikely. (You could change to `contract?.groupSlugs` if you want to update when contract becomes defined)